types.go

```diff
@@ -83,3 +61,70 @@ func (m PubKey) AsABCI() abci.PubKey {
 		Type: m.Type,
 	}
 }
+
+func (m ValidatorUpdates) Validate() error {
+	var err error
+	for _, v := range m.ValidatorUpdates {
+		err = errors.Append(err, v.Validate())
+	}
+	return err
+}
+
+// Deduplicate makes sure we only use the last validator update to any given validator.
+// For bookkeeping we have an option to drop validators with zero power, because those
+// are being remove by tendermint once propagated.
+func (m ValidatorUpdates) Deduplicate(dropZeroPower bool) []ValidatorUpdate {
+	duplicates := make(map[string]int, 0)
+	cleanValidatorSlice := make([]ValidatorUpdate, 0, len(m.ValidatorUpdates))
+
+	for _, v := range m.ValidatorUpdates {
+		if dropZeroPower && v.Power == 0 {
+			continue
+		}
+		if key, ok := duplicates[v.PubKey.String()]; ok {
+			cleanValidatorSlice[key] = v
+			continue
+		}
+		cleanValidatorSlice = append(cleanValidatorSlice, v)
+		duplicates[v.PubKey.String()] = len(cleanValidatorSlice) - 1
+	}
+
+	return cleanValidatorSlice
+}
+
+// Store stores ValidatorUpdates to the KVStore while cleaning up those with 0
+// power.
+func (m ValidatorUpdates) Store(store KVStore) error {
```

Comment:
Would it make sense to make it more weave-like and provide a bucket for this entity instead? We never implement in weave entities that can save themselves. It would be more familiar if this entity was also saved through a bucket. What do you think?

Reply:
This PR is simply a continuance of another PR in which we agreed to have a singleton storage for this as bucket is not needed.

Reply:
And the reasoning behind this storage is we only ever need one snapshot - so we don't really need a bucket here, just basically a kv store that we update on each successful tx.

Reply:
Yeah, I don't like Bucket for singleton.
Also, gconf.Configuration objects don't use Buckets either. A general prefix for all config options, but each stored separately in the kvstore.

Reply:
Also, you cannot import Bucket from top-level weave (but apparently we can import errors)

Reply:
I wanted to use a separate method initially, but I think this api is actually nicer as it allows for sort of `ActiveRecord` behaviour. 

Since this type of entity and it's storage is unprecedented in weave I see no real reason in having extra method for this as we already accept kvstore as a parameter, so there is no "magic" happening. 

I'd like to keep this as is unless this comment receives some upvotes.

Reply:
actually, disregard. we are using a method for Get, might as well use one for Store

Reply:
I like the consistency too, so it's all good with me. Will address.

Reply:
Similar problem is solved in `gconf` with [`Save`](https://github.com/iov-one/weave/blob/master/gconf/gconf.go#L16) and [`Load`](https://github.com/iov-one/weave/blob/master/gconf/gconf.go#L38) functions.

Reply:
@husio Yeah it's basically what we have here now too, but the names are somewhat more verbose, cause it's in `weave` package.

We might want to revisit this later and even provide a unified interface to store/retrieve singletons.